snap/snapdir/snapdir.go

```diff
@@ -41,6 +41,17 @@ func New(path string) *SnapDir {
 	return &SnapDir{path: path}
 }
 
+func (s *SnapDir) Size() (size int64, err error) {
+	totalSize := int64(0)
+	f := func(_ string, info os.FileInfo, err error) error {
+		totalSize += info.Size()
+		return err
+	}
+	filepath.Walk(s.path, f)
```

Comment:
I was concerned for a moment that this might cause an endless loop for symlink cycles in subdirs, but it looks like Walk() will not follow symlinks so this should be fine.